### PR TITLE
SideNav / TableOfContents fixes

### PR DESCRIPTION
**Pull request by wackerow**
## Description
Fixes various layout shifts that are occurring in dev docs, and components using the ToC component.
- Sets `initial` animation condition to match `isOpen` boolean (from "closed"); fixes bug causing Docs left desktop menu from starting in collapsed position.
- Removes use of `Show` component from `TableOfContents` and `TableOfContentsMobile`, using `hideBelow` and `hideFrom` instead; fixes browser not rendering the component on initial paint and removes the layout shift
- Patches styling applied to the desktop TableOfContents to use `lg` breakpoint.. was previously `1024px` which is larger than the `lg` breakpoint (`62em`, or `992px`) where our layout adjusts to mobile

## Related Issue
Fixes https://www.notion.so/efdn/BUG-dev-docs-sidebar-menu-is-collapsed-by-default-on-first-load-a928d191a8d74419a5375ab7c00fec54?pvs=4

## Preview link
- https://deploy-preview-213--ethereum-org-fork.netlify.app/developers/docs/blocks

### CODE REVIEW

File: src/components/TableOfContents/TableOfContentsMobile.tsx [hunk 1]
```diff
@@ -37,49 +37,47 @@ const Mobile: React.FC<IPropsTableOfContentsMobile> = ({ items, maxDepth }) => {
   }
 
   return (
-    <Show below={lgBp}>
```
Comment (wackerow) [inline src/components/TableOfContents/TableOfContentsMobile.tsx:40]: Removed `Show`

File: src/components/TableOfContents/TableOfContentsMobile.tsx [hunk 2]
```diff
@@ -37,49 +37,47 @@ const Mobile: React.FC<IPropsTableOfContentsMobile> = ({ items, maxDepth }) => {
...
-      <Box
-        as="aside"
-        background="background.base"
-        border="1px"
-        borderColor="border"
-        borderRadius="4px"
-        py={2}
-        px={4}
+    <Box
+      hideFrom={lgBp}
```
Comment (wackerow) [inline src/components/TableOfContents/TableOfContentsMobile.tsx:41]: Added `hideFrom={lgBp}`

File: src/components/TableOfContents/index.tsx [hunk 3]
```diff
@@ -68,47 +67,46 @@ const TableOfContents: React.FC<IProps> = ({
   }
 
   return (
-    <Show above={lgBp}>
```
Comment (wackerow) [inline src/components/TableOfContents/index.tsx:71]: removed `Show`

File: src/components/TableOfContents/index.tsx [hunk 4]
```diff
@@ -68,47 +67,46 @@ const TableOfContents: React.FC<IProps> = ({
...
-            <List m={0}>
-              <ItemsList
-                items={items}
-                depth={0}
-                maxDepth={maxDepth ? maxDepth : 1}
-                activeHash={activeHash}
-              />
-            </List>
+    <Box
+      hideBelow={lgBp}
```
Comment (wackerow) [inline src/components/TableOfContents/index.tsx:71]: added `hideBelow={lgBp}`

File: src/components/SideNav.tsx [hunk 5]
```diff
@@ -115,7 +115,7 @@ const NavLink: React.FC<IPropsNavLink> = ({ item, path }) => {
           key={item.id}
           animate={isOpen ? "open" : "closed"}
           variants={innerLinksVariants}
-          initial="closed"
+          initial={isOpen ? "open" : "closed"}
```
Comment (wackerow) [inline src/components/SideNav.tsx:120]: Remove default "closed", and align with logic of the `isOpen` boolean

Comment (wackerow) [reply src/components/SideNav.tsx:120]: image

Should fix any glitching when the page loads, and be open to this on first paint

Comment (pettinarip) [reply src/components/SideNav.tsx:120]: On the doc's root I still see these collapsed https://deploy-preview-213--ethereum-org-fork.netlify.app/developers/docs

**Review by pettinarip — CHANGES_REQUESTED** [review]

File: src/components/SideNav.tsx (hunk 5 above)
Comment (wackerow) [reply src/components/SideNav.tsx:120]: Hrm, okay... not sure why but this is loading differently on Netlify vs locally, will investigate

**Review by corwintines — APPROVED** [review]

**Pull request merged**